src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java

```diff
@@ -89,30 +90,30 @@
   private static final Object LOCAL_REPOSITORIES_INFO_LOCK = new Object();
 
   /**
-   * An interface to the GitHub API.
+   * A cache of local repositories.
    */
-  private GitHub github;
+  static final LRUMap<GitHubProject, LocalRepository> LOCAL_REPOSITORIES
+      = new LRUMap<>(LOCAL_REPOSITORIES_CACHE_CAPACITY, SCAN_UNTIL_REMOVABLE);
 
   /**
-   * A limited capacity cache to store the repository of a {@link GitHubProject}.
+   * Info about local repositories.
    */
-  private final GitHubDataCache<GHRepository> repositoryCache = new GitHubDataCache<>();
+  static final Map<URL, LocalRepositoryInfo> LOCAL_REPOSITORIES_INFO = new HashMap<>();
 
   /**
-   * A cache of local repositories.
+   * An interface to the GitHub API.
    */
-  final LRUMap<GitHubProject, LocalRepository> localRepositories
-      = new LRUMap<>(LOCAL_REPOSITORIES_CACHE_CAPACITY, SCAN_UNTIL_REMOVABLE);
+  private GitHub github;
 
   /**
-   * A base directory.
+   * A limited capacity cache to store the repository of a {@link GitHubProject}.
    */
-  private final Path base;
+  private final GitHubDataCache<GHRepository> repositoryCache = new GitHubDataCache<>();
 
   /**
-   * Info about local repositories.
+   * A base directory.
    */
-  private final Map<URL, LocalRepositoryInfo> localRepositoriesInfo;
+  private final Path base;
```

Comment:
The `base` path is used for loading info about local repositories. A user is allowed to pass any base directory while creating an instance of `GitHubDataFetcher`. It results to re-loading `LOCAL_REPOSITORIES_INFO` that is now static and shared between all instances of the fetcher. As a result, creating a new instance of the class with its own base path leads to unexpected updates in the other instances.

The `base` field should be removed. All instances of the fetcher should use only the default location of local repositories. The info should be initially loaded in a static block, not in a constructor. The methods, which work only with static fields, should be static.